packages/remark-stringify/lib/visitors/emphasis.js

```diff
@@ -9,8 +9,21 @@ module.exports = emphasis;
  * asterisk (`'*'`):
  *
  *     *foo*
+ *
+ * In `pedantic` mode, text which itself contains an underscore
+ * will cause the marker to default to an asterisk instead:
+ *
+ *     *foo_bar*
  */
 function emphasis(node) {
   var marker = this.options.emphasis;
-  return marker + this.all(node).join('') + marker;
+  var content = this.all(node).join('');
+
+  if (this.options.pedantic) {
+    if (marker === '_' && content.indexOf(marker) !== -1) {
+      marker = '*';
```

Comment:
I'm not sure about overriding the `marker` setting, @wooorm what do you think?

Reply:
Yeah, I wasn't sure how to feel about that, either.

I considered trying to do this fallback if and only if the `emphasis` option had been left set to the default, and skipping it if it had been manually set to `'_'`, but I wasn't sure how to feel about that, either.